ckanext/subscribe/action.py

```diff
@@ -38,13 +60,32 @@ def subscribe_signup(context, data_dict):
     :param skip_verification: Doesn't send email - instead it marks the
         subscription as verified. Can be used by sysadmins only.
         (optional, default=False)
+    :param g-recaptcha-response: Recaptcha response from frontend signup form
 
     :returns: the newly created subscription
     :rtype: dictionary
 
     '''
     model = context['model']
 
+    # Retrieve the configuration value to apply recaptcha
+    apply_recaptcha = tk.asbool(tk.config.get(
+        'ckanext.subscribe.apply_recaptcha', True
+    ))
+
+    if apply_recaptcha:
+        # Verify reCAPTCHA response
+        # Check the recaptcha response comes in as a request params
+        recaptcha_response = tk.request.params.get('g-recaptcha-response')
+        if not recaptcha_response:
+            # Check the recaptcha response comes in as a data_dict extras
+            if '__extras' in data_dict:
```

Comment:
I think we should add `g_recaptcha_response` to the [`subscribe_schema`](https://github.com/bellisk/ckanext-subscribe/blob/5da8f0639c8ebc22b37b69a6ba1b32a36d91c928/ckanext/subscribe/schema.py#L38) and then set it in the [controller](https://github.com/bellisk/ckanext-subscribe/blob/5da8f0639c8ebc22b37b69a6ba1b32a36d91c928/ckanext/subscribe/controller.py#L44) - then we wouldn't have to check the extras or the request params here. Wdyt?